### ISSUE TITLE
Emby Crashes while scanning Music Library

### ISSUE THREAD

**Issue by CountMurphy** [MediaBrowser/Emby#3391]
I can replicate this pretty much every time I tell Emby to scan my music library. Currently running version 3.5.1.0 on Arch linux. Logs are as follows:

EmbyServer.txt contains: 
`2018-07-27 19:39:39.430 Error HttpServer: Could not find handler for /embywebsocket`

Full log here: https://pastebin.com/hEiZ5CeR

There is also an unhandled exception log:
`System.IO.IOException: Unable to read data from the transport connection: Connection reset by peer. ---> System.Net.Sockets.SocketException: Connection reset by peer
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.GetResult(Int16 token)
   at System.Net.FixedSizeReader.ReadPacketAsync(Stream transport, AsyncProtocolRequest request)
   at System.Net.Security.SslState.ThrowIfExceptional()
   at System.Net.Security.SslState.CheckThrow(Boolean authSuccessCheck, Boolean shutdownCheck)
   at System.Net.Security.SslState.CheckOldKeyDecryptedData(Memory 1 buffer)
   at System.Net.Security.SslState.HandleQueuedCallback(Object& queuedStateRequest)
   at System.Net.Security.SslState.FinishHandshakeRead(Int32 newState)
   at System.Net.Security.SslState.FinishHandshake(Exception e, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.RehandshakeCompleteCallback(IAsyncResult result)
   at System.Net.LazyAsyncResult.Complete(IntPtr userToken)
   at System.Net.LazyAsyncResult.ProtectedInvokeCallback(Object result, IntPtr userToken)
   at System.Net.AsyncProtocolRequest.CompleteUserWithError(Exception e)
   at System.Net.FixedSizeReader.ReadPacketAsync(Stream transport, AsyncProtocolRequest request)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
System.IO.IOException
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.GetResult(Int16 token)
   at System.Net.FixedSizeReader.ReadPacketAsync(Stream transport, AsyncProtocolRequest request)
   at System.Net.Security.SslState.ThrowIfExceptional()
   at System.Net.Security.SslState.CheckThrow(Boolean authSuccessCheck, Boolean shutdownCheck)
   at System.Net.Security.SslState.CheckOldKeyDecryptedData(Memory1 buffer)
   at System.Net.Security.SslState.HandleQueuedCallback(Object& queuedStateRequest)
   at System.Net.Security.SslState.FinishHandshakeRead(Int32 newState)
   at System.Net.Security.SslState.FinishHandshake(Exception e, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslState.RehandshakeCompleteCallback(IAsyncResult result)
   at System.Net.LazyAsyncResult.Complete(IntPtr userToken)
   at System.Net.LazyAsyncResult.ProtectedInvokeCallback(Object result, IntPtr userToken)
   at System.Net.AsyncProtocolRequest.CompleteUserWithError(Exception e)
   at System.Net.FixedSizeReader.ReadPacketAsync(Stream transport, AsyncProtocolRequest request)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
InnerException: System.Net.Sockets.SocketException
System.Net.Sockets.SocketException (104): Connection reset by peer`

Each time I tell it to scan the music library, a new unhandled exception log is generated and emby crashes. 

What I've tried:
deleting all of my emby user data and starting over, importing my movie, and tv libraries. This was successful. Upon adding my music library again the behaviour above repeats.  Unselecting Music brainz and AudioDB for metadownload sources does not help. I am unsure what exactly the system is trying to communicate with that is failing. Unfortunately I have no other information.

**Comment by KristupasSavickas** [comment]
Are you using a termination in from of emby to add ssl? I am and I'm running into this issue when playing back media.

**Comment by KristupasSavickas** [comment]
This is in the log right before crashing
```
2018-07-29 09:05:27.684 Info HttpServer: HTTP Response 204 to 158.129.22.247. Time: 3ms. http://emby.legalize-ranch.club:8096/Sessions/Playing/Progress 
2018-07-29 09:05:38.809 Info TaskManager: IntervalTrigger fired for task: Scan media library
2018-07-29 09:05:38.809 Info TaskManager: Queueing task RefreshMediaLibraryTask
2018-07-29 09:05:38.809 Info TaskManager: Executing Scan media library
2018-07-29 09:05:38.815 Info LibraryMonitor: Stopping directory watching for path /media/tv
2018-07-29 09:05:38.816 Info LibraryMonitor: Stopping directory watching for path /media/movies
2018-07-29 09:05:38.816 Info LibraryMonitor: Stopping directory watching for path /media/music/not-organized
2018-07-29 09:05:38.820 Info App: Validating media library
2018-07-29 09:05:42.285 Info HttpClient: GET http://www.theaudiodb.com/api/v1/json/49jhsf8248yfahka89724011/album-mb.php?i=a7f852ba-08bc-36f1-92f3-4dc127f4b70a

```

**Comment by CountMurphy** [comment]
I'm not having any issues playing audio, though I am using apache to set up ssl for emby. I have not seen any error message like the one you posted. However as of this morning  unselecting Music brainz and AudioDB *appears* to be a workaround. Only disabling theaudiodb still produces the crash.

**Comment by LukePulverenti** [comment]
Resolved in 3.5.2. Thanks.

**Comment by zsmbrvr** [comment]
I've updated to 3.5.2 but I still have this issue.
